controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -83,6 +123,46 @@ bool wfa_ca::reply(Socket *sd, ieee1905_1::CmduMessageTx &cmdu_tx, const std::st
     return message_com::send_cmdu(sd, cmdu_tx);
 }
 
+bool wfa_ca::parse_params(std::unordered_map<std::string, std::string> &params,
+                          std::string &err_string)
+{
+    // create copy of all mandatory param names
+    std::list<std::string> mandatory_params;
+    for (const auto &param_name : params) {
+        mandatory_params.push_back(param_name.first);
+    }
+
+    size_t mandatory_params_cnt = 0;
+    while (!finished_tokens()) {
```

Comment:
This could just be
```
for (auto param_name : cmd_tokens_vec)
```
(assuming that that is passed as an argument to the function, obviously).

This removes the need for get_next_token() and finished_tokens() and all global variables.

Reply:
No, it doesn't fit here, because not all the elements in the vector are "param_name".
After "param_name" comes the param_value, and I need both of them in the same iteration.
I guess I could do it with for loop with an iterator and use std::next() to update the iterator, but I think it is more clear/clean the way I did it.